lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+// $FlowFixMe: Flow doesn't know about useState: https://stackoverflow.com/questions/53105954/cannot-import-usestate-because-there-is-no-usestate-export-in-react-flow-with
+import React, {useState} from 'react'
+import {Button, Modal} from 'react-bootstrap'
+
+import * as tripActions from '../../actions/trip'
+import {getComponentMessages} from '../../../common/util/config'
+import HourMinuteInput from '../HourMinuteInput'
+import type {Trip} from '../../../types'
+
+type Props = {
+  addNewTrip: typeof tripActions.addNewTrip,
+  constructNewRow: (trip: ?Trip, tripSeriesStartTime: number) => ?Trip,
+  onClose: () => void,
+  show: boolean
+}
+
+const initialState = {endTime: 0, error: '', startTime: 0, headway: 0}
+
+const TripSeriesModal = (props: Props) => {
+  const [{endTime, error, headway, startTime}, setState] = useState(initialState)
+  const messages = getComponentMessages('TripSeriesModal')
+
+  const close = () => {
+    const {onClose} = props
+    // Reset component state to default on close
+    setState({...initialState})
+    onClose()
+  }
+
+  const onClickGenerate = () => {
+    const {addNewTrip, constructNewRow} = props
+    if (startTime && endTime && headway) {
+      for (let time = startTime; time <= endTime; time += headway) {
+        addNewTrip(constructNewRow(null, time))
+      }
+      close()
+    } else {
+      setState(prevState => ({...prevState, error: 'You must provide input for all fields.'}))
+    }
```

Comment:
This feels messy. I think it would be much cleaner to disable the generate trips button when this condition isn't met. If we're worried about usability, we can add some alt text to the button that says exactly this!

This significantly reduces the amount of state handling we do while also making for better UX

Reply:
Agreed, I removed this in https://github.com/ibi-group/datatools-ui/pull/892/commits/e29b27fff2a7d41b88a3dfb1982b07101a3d2544